src/core/thread/mle_router.cpp

```diff
@@ -3863,7 +3863,7 @@ void MleRouter::HandleAddressSolicitResponse(Coap::Message *         aMessage,
     VerifyOrExit(router != nullptr);
 
     // Keep link to the parent in order to respond to Parent Requests before new link is established.
-    *router = mParent;
```

Comment:
Adding a note about this:
LGTM raises a warning about "slicing from sub-class `Parent` to base class `Router` here in the assignment". But this is correct and intended behavior here. Added `// lgtm [cpp/slicing]` to silence this warning (only for this line).

Reply:
Does explicitly casting remove the LGTM alert?

Reply:
No. I checked it. It does not unfortunately.

Reply:
What about a method that returns `Router&`?

Reply:
It may work. I can check it.
The other idea that may work is adding an intermediate pointer variable of `Router *`.

Reply:
Seems like using an intermediate pointer variable worked. 
I added `Router::SetFrom(const Parent &aParent)` method (to do the pointer ).